force-app/main/default/lwc/miscSharedJavaScript/__tests__/miscSharedJavaScript.test.js

```diff
@@ -0,0 +1,88 @@
+    afterEach(() => {
+        // The jsdom instance is shared across test cases in a single file so reset the DOM
+        while (document.body.firstChild) {
+            document.body.removeChild(document.body.firstChild);
+        }
+        // Clear mocks so that every test run has a clean implementation
+        jest.clearAllMocks();
+    });
+
+    it('calculates mortgage with default values', () => {
+        // Create initial element
+        const element = createElement('c-misc-shared-java-script', {
+            is: MiscSharedJavaScript,
+        });
+        document.body.appendChild(element);
+
+        expect(getTermOptions).toHaveBeenCalled();
+
+        const lightningButtonEl = element.shadowRoot.querySelector(
+            'lightning-button',
+        );
+        lightningButtonEl.dispatchEvent(new CustomEvent('click'));
+
+        // Return a promise to wait for any asynchronous DOM updates. Jest
+        // will automatically wait for the Promise chain to complete before
+        // ending the test and fail the test if the promise ends in the
+        // rejected state
+        return Promise.resolve().then(() => {
+            // Check if default values for principal, term, and rate are used for mortgage calculcation
+            expect(calculateMonthlyPayment).toHaveBeenCalledWith(200000, 30, 4);
+        });
+    });
+
+    it('calculates mortgage with custom values', () => {
+        // Create initial element
+        const element = createElement('c-misc-shared-java-script', {
+            is: MiscSharedJavaScript,
+        });
+        document.body.appendChild(element);
+
+        expect(getTermOptions).toHaveBeenCalled();
+
+        const lightningInputEls = element.shadowRoot.querySelectorAll(
+            'lightning-input',
+        );
+
+        lightningInputEls.forEach(el => {
+            if (el.label === 'Rate') {
+                el.value = 2;
+            } else if (el.label === 'Principal') {
+                el.value = 100000;
+            }
+            el.dispatchEvent(new CustomEvent('change'));
+        });
+
+        const lightningComboboxEl = element.shadowRoot.querySelector(
+            'lightning-combobox',
+        );
+        lightningComboboxEl.value = 15;
+        lightningComboboxEl.dispatchEvent(new CustomEvent('change'));
+
+        const lightningButtonEl = element.shadowRoot.querySelector(
+            'lightning-button',
+        );
+        lightningButtonEl.dispatchEvent(new CustomEvent('click'));
+
+        // Return a promise to wait for any asynchronous DOM updates. Jest
+        // will automatically wait for the Promise chain to complete before
+        // ending the test and fail the test if the promise ends in the
+        // rejected state
+        return Promise.resolve().then(() => {
+            // Check if default values for principal, term, and rate are used for mortgage calculcation
+            expect(calculateMonthlyPayment).toHaveBeenCalledWith(100000, 15, 2);
```

Comment:
I'd prefer these values to be `const`s defined with meaningful names and then can be used both when setting the values and in the assertion.

Reply:
Moved to `const`